packages/core/src/components/RadioButton/RadioButtonGroup.tsx

```diff
@@ -42,12 +39,12 @@ const RadioButtonGroup: React.FC<RadioButtonGroupProps> = ({
     }
   }, [defaultValue]);
 
-  const previousInitialValue = usePrevious(initialValue);
-  React.useEffect(() => {
-    if (initialValue !== previousInitialValue) {
-      onValueChange(initialValue);
+  const handleValueChange = (newValue: string) => {
+    setInternalValue(newValue);
+    if (onValueChange) {
```

Comment:
`onValueChange` is defaulted to a no-op function; I don't think we need to worry about it being undefined here right?

Reply:
Yeah, you're right. That check should be fine to remove